commit f706110d7be289eafb7df39737f4b2ef25382586
Author: Richard Hughes <richard@hughsie.com>
Date:   Wed Mar 31 16:13:42 2010 +0100

    Add a ::devices-changed signal to GpmEngine that helper code may want to use

Index: gnome-power-manager-2.31.91/src/gpm-engine.c
===================================================================
--- gnome-power-manager-2.31.91.orig/src/gpm-engine.c	2010-08-31 22:14:09.000000000 +1000
+++ gnome-power-manager-2.31.91/src/gpm-engine.c	2010-09-02 14:53:19.575141001 +1000
@@ -75,6 +75,7 @@
 	DISCHARGING,
 	LOW_CAPACITY,
 	PERHAPS_RECALL,
+	DEVICES_CHANGED,
 	LAST_SIGNAL
 };
 
@@ -507,6 +508,8 @@
 
 	gpm_engine_recalculate_state_icon (engine);
 	gpm_engine_recalculate_state_summary (engine);
+
+	g_signal_emit (engine, signals [DEVICES_CHANGED], 0);
 }
 
 /**
@@ -1220,6 +1223,13 @@
 			      G_STRUCT_OFFSET (GpmEngineClass, charge_critical),
 			      NULL, NULL, g_cclosure_marshal_VOID__POINTER,
 			      G_TYPE_NONE, 1, G_TYPE_POINTER);
+	signals [DEVICES_CHANGED] =
+		g_signal_new ("devices-changed",
+			      G_TYPE_FROM_CLASS (object_class),
+			      G_SIGNAL_RUN_LAST,
+			      G_STRUCT_OFFSET (GpmEngineClass, devices_changed),
+			      NULL, NULL, g_cclosure_marshal_VOID__VOID,
+			      G_TYPE_NONE, 0);
 }
 
 /**
Index: gnome-power-manager-2.31.91/src/gpm-engine.h
===================================================================
--- gnome-power-manager-2.31.91.orig/src/gpm-engine.h	2010-08-31 18:42:19.000000000 +1000
+++ gnome-power-manager-2.31.91/src/gpm-engine.h	2010-09-02 14:53:19.575141001 +1000
@@ -65,6 +65,7 @@
 						 UpDevice	*device);
 	void		(* discharging)		(GpmEngine	*engine,
 						 UpDevice	*device);
+	void		(* devices_changed)	(GpmEngine	*engine);
 } GpmEngineClass;
 
 GType		 gpm_engine_get_type		(void);
